src/org/zaproxy/zap/extension/alert/AlertAPI.java

```diff
@@ -0,0 +1,204 @@
+import org.parosproxy.paros.model.Model;
+import org.zaproxy.zap.extension.api.ApiException;
+import org.zaproxy.zap.extension.api.ApiImplementor;
+import org.zaproxy.zap.extension.api.ApiResponse;
+import org.zaproxy.zap.extension.api.ApiResponseElement;
+import org.zaproxy.zap.extension.api.ApiResponseList;
+import org.zaproxy.zap.extension.api.ApiResponseSet;
+import org.zaproxy.zap.extension.api.ApiView;
+
+import net.sf.json.JSONObject;
+
+public class AlertAPI extends ApiImplementor {
+
+    private static final String PREFIX = "alert";
+
+    private static final String VIEW_ALERT = "alert";
+    private static final String VIEW_ALERTS_BY_RISK = "alertsByRisk";
+    private static final String VIEW_ALERT_COUNTS_BY_RISK = "alertCountsByRisk";
+
+    private static final String PARAM_URL = "url";
+    private static final String PARAM_ID = "id";
+    private static final String PARAM_RECURSE = "recurse";
+
+    private ExtensionAlert extension = null;
+    private static final Logger logger = Logger.getLogger(AlertAPI.class);
+
+    public AlertAPI(ExtensionAlert ext) {
+        this.extension = ext;
+        this.addApiView(new ApiView(VIEW_ALERT, new String[] { PARAM_ID }));
+        this.addApiView(new ApiView(VIEW_ALERTS_BY_RISK, null, new String[] { PARAM_URL, PARAM_RECURSE }));
+        this.addApiView(new ApiView(VIEW_ALERT_COUNTS_BY_RISK, null, new String[] { PARAM_URL, PARAM_RECURSE }));
+    }
+
+    @Override
+    public String getPrefix() {
+        return PREFIX;
+    }
+
+    @Override
+    public ApiResponse handleApiView(String name, JSONObject params) throws ApiException {
+        ApiResponse result = null;
+        if (VIEW_ALERT.equals(name)) {
+            TableAlert tableAlert = Model.getSingleton().getDb().getTableAlert();
+            RecordAlert recordAlert;
+            try {
+                recordAlert = tableAlert.read(this.getParam(params, PARAM_ID, -1));
+            } catch (DatabaseException e) {
+                logger.error("Failed to read the alert from the session:", e);
+                throw new ApiException(ApiException.Type.INTERNAL_ERROR);
+            }
+            if (recordAlert == null) {
+                throw new ApiException(ApiException.Type.DOES_NOT_EXIST);
+            }
+            result = new ApiResponseElement(alertToSet(new Alert(recordAlert)));
+        } else if (VIEW_ALERTS_BY_RISK.equals(name)) {
+            String url = this.getParam(params, PARAM_URL, "");
+            boolean recurse = this.getParam(params, PARAM_RECURSE, false);
+            ApiResponseList resultList = new ApiResponseList(name);
+            result = resultList;
+
+            // 0 (RISK_INFO) -> 3 (RISK_HIGH)
+            ApiResponseList[] list = new ApiResponseList[4];
+            for (int i = 0; i < list.length; i++) {
+                list[i] = new ApiResponseList(Alert.MSG_RISK[i]);
+            }
+
+            AlertTreeModel model = extension.getTreeModel();

```

Comment:
It could reuse the same logic even more (like in core done with the `Processor`).